src/kv/store.h

```diff
@@ -394,11 +399,10 @@ namespace kv
         hash_at_snapshot = d.deserialise_raw();
       }
 
-      std::vector<Version> view_history;
-      auto c = get_consensus();
-      if (c)
+      std::vector<Version> view_history_;
+      if (view_history)
       {
-        view_history = d.deserialise_view_history();
+        view_history_ = d.deserialise_view_history();
```

Comment:
minor: can you simplify the code by passing view_history by reference (into the funtion) as you always want to return the view history and then just deserializing into view_history?

Reply:
I don't think we want to pass a reference here as we don't necessarily want to deserialise the view history. If the source KV store doesn't have a consensus, then the view history won't be included in the snapshot. This is useful to test snapshots on the KV in isolation of other components.